File: library/src/main/java/com/stealthcopter/networktools/ping/PingStats.java
package com.stealthcopter.networktools.ping;

import java.net.InetAddress;

public class PingStats {
    private final InetAddress ia;
    private final long noPings;
    private final long packetsLost;
    private final float averageTimeTaken;
    private final float minTimeTaken;
    private final float maxTimeTaken;
    private final boolean isReachable;

    public PingStats(InetAddress ia, long noPings, long packetsLost, float totalTimeTaken, float minTimeTaken, float maxTimeTaken) {
        this.ia = ia;
        this.noPings = noPings;
        this.packetsLost = packetsLost;
        this.averageTimeTaken = totalTimeTaken / noPings;
        this.minTimeTaken = minTimeTaken;
        this.maxTimeTaken = maxTimeTaken;
        this.isReachable = noPings - packetsLost > 0;
    }

    public InetAddress getAddress() {
        return ia;
    }

    public long getNoPings() {
        return noPings;
    }

    public long getPacketsLost() {
        return packetsLost;
    }

    public float getAverageTimeTaken() {
        return averageTimeTaken;
    }

    public float getMinTimeTaken() {
        return minTimeTaken;
    }

    public float getMaxTimeTaken() {
        return maxTimeTaken;
    }

    public boolean isReachable() {
        return isReachable;
    }

    public long getAverageTimeTakenMillis() {
        return (long) (averageTimeTaken * 1000);
    }

    public long getMinTimeTakenMillis() {
        return (long) (minTimeTaken * 1000);
    }

    public long getMaxTimeTakenMillis() {
        return (long) (maxTimeTaken * 1000);
    }

    @Override
    public String toString() {
        return "PingStats{" +
                "ia=" + ia +
                ", noPings=" + noPings +
                ", packetsLost=" + packetsLost +
                ", averageTimeTaken=" + averageTimeTaken +
                ", minTimeTaken=" + minTimeTaken +
                ", maxTimeTaken=" + maxTimeTaken +
                '}';
    }
}

File: library/src/main/java/com/stealthcopter/networktools/ping/PingStats.kt
package com.stealthcopter.networktools.ping

import java.net.InetAddress

class PingStats(
    val address: InetAddress,
    val noPings: Long,
    val packetsLost: Long,
    totalTimeTaken: Float,
    minTimeTaken: Float,
    maxTimeTaken: Float
) {
    val averageTimeTaken: Float
    val minTimeTaken: Float
    val maxTimeTaken: Float
    val isReachable: Boolean

    init {
        averageTimeTaken = totalTimeTaken / noPings
        this.minTimeTaken = minTimeTaken
        this.maxTimeTaken = maxTimeTaken
        isReachable = noPings - packetsLost > 0
    }

    val averageTimeTakenMillis: Long
        get() = (averageTimeTaken * 1000).toLong()
    val minTimeTakenMillis: Long
        get() = (minTimeTaken * 1000).toLong()
    val maxTimeTakenMillis: Long
        get() = (maxTimeTaken * 1000).toLong()

    override fun toString(): String {
        return "PingStats{" +
                "ia=" + address +
                ", noPings=" + noPings +
                ", packetsLost=" + packetsLost +
                ", averageTimeTaken=" + averageTimeTaken +
                ", minTimeTaken=" + minTimeTaken +
                ", maxTimeTaken=" + maxTimeTaken +
                '}'
    }
}